components/widgets/menu/menu_items/menu_item_action.jsx

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import menuItem from './menu_item.jsx';
+
+export const MenuItemActionImpl = ({onClick, text, extraText}) => (
```

Comment:
This pattern is quite clever I think. But (out of curiosity - definitely not requesting a change here) is there a reason it *has* to be done this way, or offers substantial benefits, rather than just having `FooMenuItem` being a component that has a `MenuItem` in it's render function? Really just curious to understand whether this is a clever hack to reduce code duplication, or if it actually has a different outcome over doing it another way?

Reply:
Is a quite common pattern (Higher Order Component) to reduce components repetitive logic in React Components. The point here is, if you want to define your own menu item, which have special behavior, with the "menuItem" HOC, you will have the `show`, `icon` and `id` properties working as expected, and then, you only have to build the logic of your component.

Reply:
:+1: makes sense.